src/handlers/shared/start.ts

```diff
@@ -141,12 +143,22 @@ async function handleTaskLimitChecks(username: string, context: Context, maxConc
   // check for max and enforce max
 
   if (Math.abs(assignedIssues.length - openedPullRequests.length) >= maxConcurrentTasks) {
-    const log = logger.error(username === sender ? "You have reached your max task limit" : `${username} has reached their max task limit`, {
-      assignedIssues: assignedIssues.length,
-      openedPullRequests: openedPullRequests.length,
-      maxConcurrentTasks,
-    });
-    await addCommentToIssue(context, log?.logMessage.diff as string);
+    const logMessage = username !== sender 
+      ? `${username} has reached their max task limit`
+      : null;
+    
+    const log = logMessage 
+      ? logger.error(logMessage, {
+          assignedIssues: assignedIssues.length,
+          openedPullRequests: openedPullRequests.length,
+          maxConcurrentTasks,
+        })
+      : null;
+  
+    if (log && log.logMessage.diff) {
```

Comment:
I think the code would be clearer if the whole logic was within a `if` block checking the `user === sender` to avoid the two ternaries above.

Reply:
Done